### PR TITLE
Custom amount notifications + notification tweaks

### PR DESCRIPTION
**Pull request by jwhamilton99**
Added custom amount notifications, with an amount you can set in the notification settings
Also restructured some of the significant amount code to be more efficient

### CODE REVIEW

File: Shared/Controller/NotificationSettingsController.swift [hunk 1]
```diff
@@ -25,55 +25,102 @@ class NotificationSettingsController: ObservableObject {
...
+                        }
+                    })
+                    break
+                }
+            })
+        }).store(in: &self.subscribers)
     }
     
     func removePublishers() {
-        self.subscribers = []
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:59]: Did you intend to delete this and re-add the old code?

File: Shared/Controller/NotificationSettingsController.swift [hunk 2]
```diff
@@ -25,55 +25,102 @@ class NotificationSettingsController: ObservableObject {
...
+        for subscriber in self.subscribers {
+            subscriber.cancel()
+        }
     }
     
     func refresh() {
-        if (!self.refreshing) {
+        if(!self.refreshing) {
             self.refreshing = true
+            removePublishers()
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:114]: Did you intend to re-add this?

File: Shared/Controller/NotificationSettingsController.swift [hunk 3]
```diff
@@ -25,55 +25,102 @@ class NotificationSettingsController: ObservableObject {
...
             UNUserNotificationCenter.current().getNotificationSettings(completionHandler: {(settings) in
-                
                 DispatchQueue.main.async {
                     if(settings.authorizationStatus == .authorized) {
-                        self.setupPublishers()
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:69]: Did you intend to delete this?

File: Shared/Controller/NotificationSettingsController.swift [hunk 4]
```diff
@@ -25,55 +25,102 @@ class NotificationSettingsController: ObservableObject {
...
                         self.showMilestones = UserDefaults.shared.showMilestoneNotification
                         self.showGoal = UserDefaults.shared.showGoalNotification
                         self.showSignificantAmounts = UserDefaults.shared.showSignificantAmountNotification
                         self.showMilestoneAdded = UserDefaults.shared.showMilestoneAddedNotification
                         self.notificationsAllowed = true
                     } else {
-                        self.removePublishers()
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:76]: Did you intend to remove this?

File: Shared/Controller/NotificationSettingsController.swift [hunk 5]
```diff
@@ -85,6 +132,12 @@ class NotificationSettingsController: ObservableObject {
                         self.notificationAccessAsked = true
                     }
                     
+                    self.customAmountInput = formatCurrency(from: String((UserDefaults.shared.object(forKey: "customNotificationAmount") as? Double) ?? -1.0), currency: "USD", showFullCurrencySymbol: false).1
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:135]: Might what do use `UserDefaults.shared.double`?
(https://developer.apple.com/documentation/foundation/userdefaults/1416581-double)

File: Shared/Controller/NotificationSettingsController.swift [hunk 6]
```diff
@@ -25,55 +25,102 @@ class NotificationSettingsController: ObservableObject {
...
+    @Published var enableCustomAmountNotification: Bool = false
+    @Published var customAmountInput: String = ""
+    
+    @Published var rejectedInputShowing: Bool = false
+    
     init() {
-        self.showMilestones = UserDefaults.shared.showMilestoneNotification
-        self.showGoal = UserDefaults.shared.showGoalNotification
-        self.showSignificantAmounts = UserDefaults.shared.showSignificantAmountNotification
-        self.showMilestoneAdded = UserDefaults.shared.showMilestoneAddedNotification
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:32]: Did you intend to remove all of these?

**Review by dwrss — CHANGES_REQUESTED** [review]
I think there might have been an issue merging/rebasing from main as this reverts a bunch of changes?

**Comment by jwhamilton99** [comment]
Fixed pull request in #44, this one can be closed.

**Pull request closed without merging**